tools/IndexStore/CMakeLists.txt

```diff
@@ -49,6 +49,10 @@ if(APPLE)
 
   set_property(TARGET IndexStore APPEND_STRING PROPERTY
                LINK_FLAGS ${INDEXSTORE_LINK_FLAGS})
+else()
+  # Blocks are off by default on non-darwin platforms.
+  target_compile_options(IndexStore PRIVATE -fblocks)
+  target_link_libraries(IndexStore PRIVATE BlocksRuntime)
```

Comment:
Ugh, this is really annoying.  The `BlocksRuntime` really should come from libdispatch, as that is the canonical one that is used elsewhere in the swift project.  I don't know how to break this cycle.